src/to_wkt/geo_trait_impl.rs

```diff
@@ -0,0 +1,488 @@
+use core::fmt;
+use std::fmt::Write;
+
+use geo_traits::{
+    CoordTrait, GeometryCollectionTrait, GeometryTrait, LineStringTrait, LineTrait,
+    MultiLineStringTrait, MultiPointTrait, MultiPolygonTrait, PointTrait, PolygonTrait, RectTrait,
+    TriangleTrait,
+};
+
+use crate::error::Error;
+use crate::types::Coord;
+use crate::WktNum;
+
+/// The physical size of the coordinate dimension
+///
+/// This is used so that we don't have to call `.dim()` on **every** coordinate. We infer it once
+/// from the `geo_traits::Dimensions` and then pass it to each coordinate.
+#[derive(Clone, Copy)]
+enum PhysicalCoordinateDimension {
+    Two,
+    Three,
+    Four,
+}
+
+impl From<geo_traits::Dimensions> for PhysicalCoordinateDimension {
+    fn from(value: geo_traits::Dimensions) -> Self {
+        match value.size() {
+            2 => Self::Two,
+            3 => Self::Three,
+            4 => Self::Four,
+            size => panic!("Unexpected dimension for coordinate: {}", size),
+        }
+    }
+}
+
+pub fn write_point<T: WktNum + fmt::Display, G: PointTrait<T = T>, W: Write>(
+    g: &G,
+    f: &mut W,
+) -> Result<(), Error> {
+    let dim = g.dim();
+    // Write prefix
+    match dim {
+        geo_traits::Dimensions::Xy | geo_traits::Dimensions::Unknown(2) => f.write_str("POINT"),
+        geo_traits::Dimensions::Xyz | geo_traits::Dimensions::Unknown(3) => f.write_str("POINT Z"),
+        geo_traits::Dimensions::Xym => f.write_str("POINT M"),
+        geo_traits::Dimensions::Xyzm | geo_traits::Dimensions::Unknown(4) => {
+            f.write_str("POINT ZM")
+        }
+        geo_traits::Dimensions::Unknown(_) => return Err(Error::UnknownDimension),
+    }?;
+    let size = PhysicalCoordinateDimension::from(dim);
+    if let Some(coord) = g.coord() {
+        f.write_char('(')?;
+        add_coord(&coord, f, size)?;
+        f.write_char(')')?;
+        Ok(())
+    } else {
+        Ok(f.write_str(" EMPTY")?)
+    }
+}
+
+pub fn write_linestring<T: WktNum + fmt::Display, G: LineStringTrait<T = T>, W: Write>(
+    linestring: &G,
+    f: &mut W,
+) -> Result<(), crate::error::Error> {
```

Comment:
```suggestion
) -> Result<(), Error> {
```

Error is already in scope (similar throughout this file)